.github/workflows/publish-gh-pages.yml

```diff
@@ -0,0 +1,97 @@
+# Sample workflow for building and deploying a Jekyll site to GitHub Pages
+name: Publish site
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["gh-pages"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: gh-pages
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v3
+
+      - name: Build with Jekyll
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./
+          destination: ./_site
+
+      - name: Download C apidocs artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          name: onnxruntime-c-apidocs
+          workflow: publish-c-apidocs.yml
+          path: apidocs
+
+      - name: Download C# apidocs artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          name: onnxruntime-csharp-apidocs
+          workflow: publish-csharp-apidocs.yml
+          path: apidocs
+
+      - name: Download Java apidocs artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          name: onnxruntime-java-apidocs
+          workflow: publish-java-apidocs.yml
+          path: apidocs
+
+      - name: Download Python apidocs artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          name: onnxruntime-python-apidocs
+          workflow: publish-python-apidocs.yml
+          path: apidocs
+
+      - name: Move apidocs folder into place
+        run: |
+          sudo rm -rf _site/docs/api/c
```

Comment:
do we need to use `sudo`?

Reply:
Yes, needed for both commands. I experimented with all combos here: https://github.com/natke/onnxruntime/actions/runs/4492983164/jobs/7903602815